src/meshes/t8code_mesh.jl

```diff
@@ -293,18 +394,38 @@ conforming mesh from a `p4est_connectivity` data structure.
 - `RealT::Type`: the type that should be used for coordinates.
 - `initial_refinement_level::Integer`: refine the mesh uniformly to this level before the simulation starts.
 """
-function T8codeMesh{NDIMS}(conn::Ptr{p4est_connectivity}; kwargs...) where {NDIMS}
-    @assert NDIMS == 2 # Only support for NDIMS = 2 yet.
-
+function T8codeMesh(conn::Ptr{p4est_connectivity}; kwargs...)
```

Comment:
Does the old version `T8codeMesh{NDIMS}(...)` `still work? If not, this is a breaking change.

Reply:
No they don't. Indeed this is a breaking change.

Reply:
To avoid this breaking change, could you please deprecate the old versions?

Reply:
How would this have to be done? Using something like `Base.@deprecate`? Do we have a workflow or guideline in Trixi.jl for deprecating things?

Reply:
Added `@deprecated` calls to `src/meshes/t8code_mesh.jl`.

I did not find any precedence elsewherre in Trixi.